### PR TITLE
feat: refactor and improve the request client and support refreshToken

### LINKED ISSUE #4121: FEATURE: Support refresh token

### PR DESCRIPTION
**Pull request by likui628**
## Description

closes #4121

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

## Checklist

> ℹ️ Check all checkboxes - this will indicate that you have done everything in accordance with the rules in [CONTRIBUTING](contributing.md).

- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs:dev` command.
- [ ] Run the tests with `pnpm test`.
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Summary by CodeRabbit

- **New Features**
	- Introduced improved authentication flow with enhanced token management, including access and refresh tokens.
	- Added logout and refresh token functionality, allowing users to effectively manage their sessions.
	- New utilities for handling cookies and JWT, enhancing security and efficiency.
	- Added new interfaces and asynchronous functions for handling token-related API calls.
	- Enhanced request client with interceptors for better error handling and authorization.

- **Bug Fixes**
	- Enhanced error handling in authentication processes, providing clearer responses for unauthorized access.

- **Documentation**
	- Updated API responses and token management documentation for clarity, including new methods for token refreshing.

- **Chores**
	- Updated dependencies to improve security and maintainability.

### CODE REVIEW

File: apps/web-antd/src/api/request.ts [hunk 1]
```diff
@@ -61,7 +47,30 @@ function createRequestClient(baseURL: string) {
     }
     throw new Error(`Error ${status}: ${msg}`);
   });
+
+  // todo 这个拦截器需要在request-client内默认response interceptor之前注册
```
Comment (anncwb) [inline apps/web-antd/src/api/request.ts:51]: 多个addResponseInterceptor 可能得需要重新重构下Request，支持排序或者全部归纳到一个拦截器就好

File: apps/web-antd/src/layouts/basic.vue [hunk 2]
```diff
@@ -137,16 +129,6 @@ function handleMakeAll() {
         @make-all="handleMakeAll"
       />
     </template>
-    <template #extra>
```
Comment (anncwb) [inline apps/web-antd/src/layouts/basic.vue:140]: 这个是删除了吗

Comment (likui628) [reply apps/web-antd/src/layouts/basic.vue:140]: 无感刷新了，登录的对话框感觉没必要了

Comment (likui628) [reply apps/web-antd/src/layouts/basic.vue:140]: 我后面看下，尽量保留吧

**Review by anncwb — APPROVED** [review]

**Pull request merged**